apps/chat/src/store/models/models.reducers.ts

```diff
@@ -237,6 +251,20 @@ export const modelsSlice = createSlice({
       );
       state.modelsMap = omit(state.modelsMap, payload.references);
     },
+    deleteSharedWithMeModel: (
+      state,
+      { payload }: PayloadAction<{ modelId: string }>,
+    ) => {
+      const modelReference = state.modelsMap[payload.modelId]?.reference;
+
+      state.models = state.models.filter(
+        (model) => model.id !== payload.modelId,
+      );
+      state.recentModelsIds = state.recentModelsIds.filter(
+        (id) => id !== modelReference,
+      );
+      state.modelsMap = omit(state.modelsMap, payload.modelId);
```

Comment:
also need to remove by reference-key
```suggestion
      state.modelsMap = omit(state.modelsMap, [payload.modelId, modelReference]);
```

Reply:
Thank you for the suggestion!
This part was changed.